src/main/java/org/terasology/climateConditions/HypothermiaSystem.java

```diff
@@ -0,0 +1,110 @@
+        //TODO: react on OnBiomeChangedEvent to handle the danger zone
+        float height = event.getPosition().getY();
+        float lastHeight = height - event.getDelta().getY();
+        if (height > thresholdHeight && lastHeight <= thresholdHeight) {
+            player.addOrSaveComponent(new HypothermiaComponent());
+        }
+        if (height < thresholdHeight && lastHeight >= thresholdHeight) {
+            if(player.hasComponent(HypothermiaComponent.class)){
+                player.removeComponent(HypothermiaComponent.class);
+            }
+        }
+    }
+
+    @ReceiveEvent
+    public void onHypothermia(OnAddedComponent event, EntityRef player, HypothermiaComponent hypothermia){
+        if(player.hasComponent(HypothermiaComponent.class)){
+            delayManager.addPeriodicAction(player, VISIBLE_BREATH_ACTION_ID, initialDelay, breathInterval);
+        }
+    }
+
+    @ReceiveEvent(components = {LocationComponent.class})
+    public void onPeriodicBreath(PeriodicActionTriggeredEvent event, EntityRef player) {
+        if (event.getActionId().equals(VISIBLE_BREATH_ACTION_ID)) {
+            // Check to see if breath should be visible
+            LocationComponent location = player.getComponent(LocationComponent.class);
+            float height = location.getLocalPosition().getY();
+            if (height > thresholdHeight) {
+                updateVisibleBreathEffect(player);
+            }
+        }
+    }
+    private void updateVisibleBreathEffect(EntityRef player) {
+        EntityRef particleEntity = entityManager.create("climateConditions:VisibleBreathEffect");
+        LocationComponent targetLoc = player.getComponent(LocationComponent.class);
+        LocationComponent childLoc = particleEntity.getComponent(LocationComponent.class);
+        childLoc.setWorldPosition(targetLoc.getWorldPosition());
+        Location.attachChild(player, particleEntity);
+        particleEntity.setOwner(player);
+        Vector3f direction = targetLoc.getLocalDirection();
+        direction.normalize();
+
+        //NOTE: initializing velocity using the constructor wasn't working hence the long method.
+        //particleEntity.addComponent(new VelocityRangeGeneratorComponent(direction,direction.scale(2)));
+        VelocityRangeGeneratorComponent velocity = particleEntity.getComponent(VelocityRangeGeneratorComponent.class);
+        direction.scale((float) 0.5);
+        direction.addY((float) 0.5);
+        velocity.minVelocity = direction;
+        direction.scale((float) 1.5);
+        velocity.maxVelocity = direction;
+        particleEntity.addOrSaveComponent(velocity);
+        if (!player.hasComponent(VisibleBreathComponent.class)) {
+            player.addComponent(new VisibleBreathComponent());
+        }
+        player.getComponent(VisibleBreathComponent.class).particleEntity = particleEntity;
+    }
```

Comment:
I think both `onPeriodicBreath` and `updateVisibleBreathEffect` would be better off it their own system `VisibleBreathingSystem`. The action id `VISIBLE_BREATH_ACTION_ID` should also be in that system.

Reason being is that although it is triggered by hypothermia, applying the effect itself is unrelated to that condition. Thus, we can separate the concerns early on here.